feature-libs/cart/saved-cart/components/saved-cart-form-dialog/saved-cart-form-dialog.component.spec.ts

```diff
@@ -80,5 +113,133 @@ describe('SavedCartFormDialogComponent', () => {
+    });
+  });
+
+  describe('should call service on save cart', () => {
+    beforeEach(() => {
+      spyOn(savedCartService, 'saveCart');
+    });
+
+    it('with filled values', () => {
+      mockDialogData$.next(mockFilledDialogData);
+      fixture.detectChanges();
+      component.saveOrEditCart(mockCartId);
+
+      expect(savedCartService.saveCart).toHaveBeenCalledWith({
+        cartId: mockCartId,
+        saveCartName: mockCart.name,
+        saveCartDescription: mockCart.description,
+        extraData: { edit: true },
+      });
+    });
+
+    it('with empty values', () => {
+      mockDialogData$.next(mockEmptyDialogData);
+      fixture.detectChanges();
+      component.saveOrEditCart(mockCartId);
+
+      expect(savedCartService.saveCart).toHaveBeenCalledWith({
+        cartId: mockCartId,
+        saveCartName: '',
+        saveCartDescription: '',
+        extraData: { edit: false },
+      });
+    });
+  });
+
+  it('should apply multiple actions on delete cart', () => {
+    spyOn(routingService, 'go');
+    spyOn(globalMessageService, 'add');
+    spyOn(savedCartService, 'deleteSavedCart');
+    spyOn(component, 'close');
+
+    component.deleteCart(mockCartId);
+
+    expect(routingService.go).toHaveBeenCalledWith({
+      cxRoute: 'savedCarts',
+    });
+    expect(globalMessageService.add).toHaveBeenCalledWith(
+      mockDeleteGlobalMessageObject,
+      GlobalMessageType.MSG_TYPE_CONFIRMATION
+    );
+    expect(savedCartService.deleteSavedCart).toHaveBeenCalledWith(mockCartId);
+    expect(component.close).toHaveBeenCalledWith(mockSuccessDeleteCloseReason);
+  });
+
+  it('should close dialog on close method', () => {
+    spyOn(launchDialogService, 'closeDialog');
+    component.close(mockSuccessDeleteCloseReason);
+
+    expect(launchDialogService.closeDialog).toHaveBeenCalledWith(
+      mockSuccessDeleteCloseReason
+    );
+  });
+
+  describe('should return actual characters left', () => {
+    it('when form control has value', () => {
+      component?.form?.get('description')?.setValue('test');
+
+      expect(component.descriptionsCharacterLeft).toEqual(
+        mockDescriptionMaxLength - 4
+      );
+    });
+
+    it('when form control has null value', () => {
+      component?.form?.get('description')?.setValue(null);
+
+      expect(component.descriptionsCharacterLeft).toEqual(
+        mockDescriptionMaxLength
+      );
+    });
+
+    it('when form control has undefined value', () => {
+      component?.form?.get('description')?.setValue(undefined);
+
+      expect(component.descriptionsCharacterLeft).toEqual(
+        mockDescriptionMaxLength
+      );
+    });
+    it('when form control has empty value', () => {
+      component?.form?.get('description')?.setValue('');
+
+      expect(component.descriptionsCharacterLeft).toEqual(
+        mockDescriptionMaxLength
+      );
+    });
+  });
+
+  //  TODO
+  // describe('watch save cart process success', () => {
```

Comment:
I left the comment in the DM regarding this to not forget this

Did we agree, that we will test this from the call on Tuesday?

Reply:
Someone can approve and you can merge it in afterwards